plugins/action_send_notification/src/sendNotification.ts

```diff
@@ -124,9 +124,15 @@ const sendNotification: ActionPluginType = async ({ parameters, applicationData,
           )
           db.notificationEmailError(notificationResult.id, `ERROR: ${err.message}`)
         })
-    } else {
+    }
+    //warnings and errors if send email isn't working
+    if (!transporter) {
       console.log('Email not sent - missing email configuration')
-      db.notificationEmailError(notificationResult.id, `WARNING: email configuration not provided`)
+      db.notificationEmailError(notificationResult.id, `WARNING: Email configuration not provided`)
+    }
+    if (!hasValidEmails) {
+      console.log('Email not sent - no valid email address')
+      db.notificationEmailError(notificationResult.id, `WARNING: No valid email addresses`)
```

Comment:
Problem here is that if `!transporter` then we're likely to also have `!hasValidEmails`. In which case the second one will just overwrite the first one.

Would suggest `else if` so it only checks the second one if the first condition isn't met:
```ts
if (!transporter) {
      console.log('Email not sent - missing email configuration')
      db.notificationEmailError(notificationResult.id, `WARNING: Email configuration not provided`)
    } else if (!hasValidEmails) {
      console.log('Email not sent - no valid email address')
      db.notificationEmailError(notificationResult.id, `WARNING: No valid email addresses`)
    }
```